### PR TITLE
feat(ans104): index data item data hashes

### PR DESCRIPTION
**Pull request by karlprieb**

### CODE REVIEW

File: src/workers/ans104-data-indexer.ts [hunk 1]
```diff
@@ -83,12 +92,24 @@ export class Ans104DataIndexer {
         typeof item.data_size === 'number'
       ) {
         log.debug('Indexing data item data...');
+        await this.contiguousDataIndex.saveDataContentAttributes({
+          id: item.id,
+          hash: item.data_hash,
+          dataSize: item.data_size,
+          contentType: item.content_type,
+          cachedAt: currentUnixTimestamp(),
```
Comment (djwhitt) [inline src/workers/ans104-data-indexer.ts:100]: Let's drop `cachedAt` here. We shouldn't include it unless we're actually writing the data to the cache. I may have been the one who said we needed it too, my bad is so.

**Review by djwhitt — CHANGES_REQUESTED** [review]
Nice work! I requested a couple small changes, but nothing major.

File: src/lib/ans-104.ts [hunk 2]
```diff
@@ -62,26 +63,47 @@ interface ParserMessage {
...
 }): NormalizedDataItem {
+  let contentType: string | undefined;
   const tags = (ans104DataItem.tags || []).map(
-    (tag: { name: string; value: string }) => ({
-      name: utf8ToB64Url(tag.name),
-      value: utf8ToB64Url(tag.value),
-    }),
+    (tag: { name: string; value: string }) => {
+      if (tag.name.toLowerCase() === 'content-type') {
+        contentType = tag.value;
```
[the remark this answers is not in the document]
Comment (karlprieb) [reply src/lib/ans-104.ts:104]: [`e1e3da6` (#102)](https://github.com/ar-io/ar-io-node/pull/102/commits/e1e3da6af5a449a59940d958a21dae11d3629329)

File: src/workers/ans104-data-indexer.ts (hunk 1 above)
Comment (karlprieb) [reply src/workers/ans104-data-indexer.ts]: [`b20b953` (#102)](https://github.com/ar-io/ar-io-node/pull/102/commits/b20b953ce83da4395881be89d71b14eae724d09c)

**Review by djwhitt — APPROVED** [review]

**Pull request merged**